SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h

```diff
@@ -504,6 +507,19 @@ const TArray<Worker_ComponentId> KnownEntityAuthorityComponents = { POSITION_COM
 																	GSM_SHUTDOWN_COMPONENT_ID,	 VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID,
 																	SERVER_WORKER_COMPONENT_ID };
 
+// SPATIAL_SNAPSHOT_VERSION is the current version of supported snapshots.
+//
+// If changes are made to the schema or content of snapshots then:
+//	(a) (optional) if the schema has changed, update SPATIAL_SNAPSHOT_SCHEMA_HASH,
```

Comment:
Snapshots can become invalid for one or more reasons, e.g. changing schema, adding entities, etc. 

This PR only validates schema changes, other validation will be added later.

Therefore, if you change schema you should do parts (a) and (b), however if you add entities to a default snapshot (say) but don't change the schema you should do part (b) only.

Reply:
That makes sense, maybe add that to the comment?

Reply:
Additionally I think it would make sense to move the hash generating code here or in a related helper file it feels a bit odd encapsulate that into a test. Maybe it should even be a separate script / helper utility all together but perhaps that is over engineering it.

Reply:
Thinking about it its pretty minor at this point, maybe more for future reference I think eventually it would be nice to have an external script for generating hashes that maybe even can be automated as part of the build script?

Reply:
yup definitely, good points, this is only step 1.